src/bq769x0/bms_bq769x0.c

```diff
@@ -548,15 +550,15 @@ void bms_handle_errors(BmsConfig *conf, BmsStatus *status)
     static uint32_t sec_since_error = 0;
 
     // ToDo: Handle also temperature and chg errors (incl. temp hysteresis)
+    SYS_STAT_Type sys_stat;
+    sys_stat.byte = bq769x0_read_byte(BQ769X0_SYS_STAT);
+    error_status = sys_stat.byte;
 
     if (!bq769x0_alert_flag() && error_status == 0) {
         return;
     }
     else {
 
-        SYS_STAT_Type sys_stat;
-        sys_stat.byte = bq769x0_read_byte(BQ769X0_SYS_STAT);
-
```

Comment:
This fix is an issue if you have uv_limt=uv_reset (which i had first), the routine try to reset overvoltage flag in BQ769x0, but the overvoltage is still there. No new interrupt is rised, because SYS_STAT is unchanged. So the overvoltage flag persist and is not cleaned because bms_handle_errors is not executed again. I measured signal ALERT, it is high, stays high. But no interrupt is rised. MOSFET never switches.

Reply:
Ah, ok, makes sense. If the ALERT pin stays high we should probably also not check for the interrupt flag, but for the pin state instead?

In any case your implementation is more safe and should fix the issue you described, so let's use that.